### DISCUSSION TITLE
In Raspberry Pi 4 it works great

### DISCUSSION THREAD

**Discussion by sunilsankar** [devlikeapro/waha#123] [Show and tell]
@allburov amazing lib. I was looking for options to send alerts of my home automation to my WhatsApp and  waha core helped me solve this. I was able send my Prometheus alerts to my WhatsApp and earlier I used signal-cli ,was also looking for alternative , so now my alerts are sent to both WhatsApp and signal. 
 
Just wanted share my experience 😁

**Comment by allburov** [comment]
Hi! Thank you for your kind words, it really means a lot for us!
I'm glad that project's been used not only for spam, but for home automation as well! :heart: 
Amazing. looking forward to read an article about your system and how it works! :smile_cat:

**Comment by devlikepro** [comment]
Anyone who has problems with Raspberry like error:
> docker: no matching manifest for linux/arm64/v8 in the manifest list entries

try using 
```
devlikeapro/waha-plus:arm	
```
or 
```
devlikeapro/waha:arm	
```
images 
https://waha.devlike.pro/docs/how-to/engines/#docker-images

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by darlinton** [reply]
it is not working on raspberry 4

```
pi@raspberrypi:~ $ docker pull devlikeapro/waha:arm
arm: Pulling from devlikeapro/waha
no matching manifest for linux/arm/v7 in the manifest list entries
```

**Reply by devlikepro** [reply]
@darlinton hi! 
Could you create [a feature request](https://github.com/devlikeapro/waha/issues) for that? 

Could you also try to build it locally  in pi4 and report in the feature request the result (success/or not)
```bash
git clone https://github.com/devlikeapro/waha.git
docker build -t devlikeapro/waha:arm .
```

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by devlikepro** [reply]
> it is not working on raspberry 4

What CPU/OS does it have (not expert in rapsberri ecosystem, what it could have) ? Looks like 32 bit smthg. We're building 64bit image, but probably could also add 32bit, just need to have more info

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by sunilsankar** [reply]
Just tested in pi4 the image is working 
```
docker pull devlikeapro/waha:arm
```

Please paste the output of this command
```
uname -m
```

**Reply by darlinton** [reply]
Thanks for the fast reply.  Yes, the OS version is 32-bit. 

pi@raspberrypi:~ $ uname -a
Linux raspberrypi 5.10.103-v7l+ #1529 SMP Tue Mar 8 12:24:00 GMT 2022 armv7l GNU/Linux

Here it is the specs:

pi@raspberrypi:~ $ cat /proc/cpuinfo
processor	: 0
model name	: ARMv7 Processor rev 3 (v7l)
BogoMIPS	: 108.00
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

... other 3 processors ...

Hardware	: BCM2711
Revision	: d03114
Serial		: 100000007ff39228
Model		: Raspberry Pi 4 Model B Rev 1.4

pi@raspberrypi:~ $ uname -m
armv7l

pi@raspberrypi:~ $ lsb_release -a
No LSB modules are available.
Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux 10 (buster)
Release:	10
Codename:	buster

**Reply by sunilsankar** [reply]
Please install 64 bit os and try . It should work

**Reply by devlikepro** [reply]
> armv7l is 32 bit processor.

Is it possible to install 64bit os on 32 bit CPU :thinking:

@darlinton please try building docker image on this and see if it works! 
https://github.com/devlikeapro/waha/discussions/123#discussioncomment-12438050

If it does - kindly create a feature request and we'll see what we can do to publish 32 bit image (rare case tho)

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by sunilsankar** [reply]
It is raspberry pi 4 and it supports 64 bit os. Now all the arm hardwares including PI zero support 64bit . @devlikepro why do you want to build on 32 bit

**Reply by devlikepro** [reply]
@sunilsankar as I can see from logs it's **ARMv7** which is **32bit** **CPU**.

I'm not an expert in ARM architecture (neither CPU at all), but in x86 you can not run 64bit OS on 32bit CPU.

> Now all the arm hardwares including PI zero support 64bit

That's good, but what if you have old one? 

> why do you want to build on 32 bit

It'd costs us nothing if it's just one line in github actions, why not.

Let's wait @darlinton feedback and see, may be 32bit doesn't support other tools (like docker/nodejs/whatever we're using) so it won't be even possible to publish 32bit image.

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by sunilsankar** [reply]
Only PI1 is 32 bit from PI 2 it supports 64 bit so  32 bit in arm is waste of effort for you
